sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -1173,9 +1185,14 @@ private class BulkWriter(container: CosmosAsyncContainer,
           returnValue = writeConfig.itemWriteStrategy match {
               case ItemWriteStrategy.ItemBulkUpdate =>
                   this.shouldRetryForItemPatchBulkUpdate(statusCode, subStatusCode)
-              case _ => Exceptions.canBeTransientFailure(statusCode, subStatusCode) ||
-                statusCode == 0 // Gateway mode reports inability to connect due to
-                                // PoolAcquirePendingLimitException as status code 0
+              // Upsert can return 404/0 in rare cases (when due to TTL expiration there is a race condition
```

Comment:
should this also change be added in pointWriter as well?

Reply:
not sure whether this could happen for point operations
But won't hurt - so, will add it there as well.

Reply:
Fixed